### ISSUE TITLE
emacs-mode keybindings similar to eww?

### ISSUE THREAD

**Issue by divansantana** [atlas-engineer/nyxt#1118]
What do you think about emacs keybindings to be made similar to EWW?

Something like:

```
b 'bookmark-page
s 'switch-buffer
q 'delete-current-buffer
g 'reload-buffer
o 'follow-hint
O 'follow-hint-new-buffer
p 'switch-buffer-previous
n 'switch-buffer-next
p 'history-backwards
r 'history-forwards
```

Probably depends on #1020 .

**Comment by jmercouris** [comment]
an interesting idea, the bindings would then have to be modal though. as you said, it depends on 1020. I wouldn't be opposed to making the keybindings operate in this fashion, I would still like to prefix them though. Maybe with Control.

**Comment by jmercouris** [comment]
Thank you for the suggestion!

**Comment by aadcg** [comment]
@jmercouris why would you prefer to prefix the keybindings?

**Comment by divansantana** [comment]
Anyone manage to configure eww like bindings, perhaps by overriding the vim mode?

**Comment by aadcg** [comment]
@divansantana, I still didn't take a deeper look. But, since I'm really interested in this, I'll surely come back with something.

**Comment by divansantana** [comment]
```org-mode
|                   | eww         |
|-------------------+-------------|
| open url          | G           |
| new tab           | M-RET       |
| close tab         | q           |
| tab next          |             |
| tab previous      |             |
| back              | l           |
| forward           | r           |
| switch to buffer  | s           |
| reload            | g           |
| page down         | DEL / S-SPC |
| page up           | M-v         |
| password complete |             |
| hint open         |             |
| hint new tab      |             |
| undo              |             |
| edit in emacs     |             |
| copy url          | w           |
| scroll down       | DEL / S-SPC |
| scroll up         |             |
| search            | C-s         |
| search next       | C-s         |
| search prev       | C-r         |
| M-x               | M-x         |
| open in mpv       |             |
| begin buffer      | <           |
| end buffer        | >           |
| list bookmarks    | B           |
| list history      | H           |
| reabable mode     | R           |
| list buffers      | S           |
| add bookmark      | b           |
| download          | d           |
| next url          | n           |
| previous url      | p           |
| top url           | t           |
| up url            | u           |
| view source       | v           |
| toggle colours    | M-C         |
| copy for org mode | C-c C-x C-w |
```

**Comment by jmercouris** [comment]
Prefixing them with control causes many issues. For example C-g would be what a emacs user would expect to cancel, but that would set the URL. I think more people are familiar with general Emacs bindings than EWW specific ones. One can always make an EWW scheme and supply it. If you are willing to define this keymap and make the schemes for it, I would be willing to merge it in.

**Comment by aadcg** [comment]
@jmercouris, indeed an eww keymap only makes sense for modal keymaps. Nyxt's `emacs-mode` isn't modal, the reasons being clear. 

Therefore, oddly enough, an eww-inspired keymap would actually be akin to Nyxt's `vi-mode` (since it distinguishes between normal/insert modes). [Personally, this doesn't itch me enough. Maybe one day it will, or it might itch others](https://nyxt.atlas.engineer/article/the-thin-line-between-users-and-collaborators.org) :)

**Comment by jmercouris** [comment]
If you wish to make an EWW keymap, please open a new issue with that information. For now, I am closing this.
